Fix sed expression in test script on macos

Baltoli opened the pull request:
The behaviour of sed expressions can be slightly different across
platforms - the new expression in this PR achieves the same effect
(substituting the long Source expression on the first line for
Source...), but works on macos.

cos commented on java-backend/src/main/java/org/kframework/backend/java/symbolic/FastRuleMatcher.java:
```diff
@@ -109,6 +110,11 @@ public FastRuleMatcher(GlobalContext global, int ruleCount) {
+            // skip over IO rules when in prove rules
+            if (proveFlag && rule.containsAttribute("stream")) {
+                continue;
+            }
```
added #2189 to track

cos commented on k-distribution/tutorial/1_k/2_imp/lesson_4/imp.k:
```diff
@@ -54,4 +54,8 @@ module IMP
+
+// verification ids
+  syntax Id ::= "n"     [token]
+              | "sum"   [token]
```
could we declare these somewhere else?

andreistefanescu replied:
it should be possible to declare them in the same module as the specification rules, but then that would require the creation of a new parser form that module, so for now I would leave them here.

cos replied:
ok

Baltoli committed:
Fix sed expression on macos

The behaviour of sed expressions can be slightly different across
platforms - the new expression in this PR achieves the same effect
(substituting the long Source expression on the first line for
Source...), but works on macos.
